### DISCUSSION TITLE
Getting the ACCOUNT_THUMBPRINT into haproxy.cfg

### DISCUSSION THREAD

**Discussion by tuck1s** [acmesh-official/acme.sh#5096] [Q&A]
I started from [this tutorial](https://www.haproxy.com/blog/haproxy-and-let-s-encrypt) which explains the advantages of using acme.sh for haproxy, i.e.

* works well with TLS SNI, can have many different certs in a directory
* Puts the cert/key combined.pem file in the right place
* Does a "hot update" of haproxy with no need to restart the service (important for service continuity)

One slightly tricky thing is: how to get the `ACCOUNT_THUMBPRINT` out afterward, i.e. beyond the point when initial account registration is done? (It's in backscroll and not easy to automate).
It would be nice to automate that, possibly using `sed -i` to update my haproxy.cfg.

**Comment by aleixq** [comment]
I am running on openwrt and it's just: 
`/usr/lib/acme/client/acme.sh --register-account --accountemail [EMAIL_USED] --server letsencrypt --home /etc/acme`  
adapt to your needs

will return something like

```
[Thu Jul  4 14:40:18 UTC 2024] Registering account: https://acme-v02.api.letsencrypt.org/directory
[Thu Jul  4 14:40:19 UTC 2024] Already registered
[Thu Jul  4 14:40:19 UTC 2024] ACCOUNT_THUMBPRINT='*****YOUR_THUMBPRINT*****'
```
